### PR TITLE
Changes to add selected video effects when camera is off in Composites

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
Changes to add selected video effects when camera is off in Composites

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3193155

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 1]
```diff
@@ -121,6 +121,11 @@ export type CallAdapterClientState = {
    * Default set of background images for background replacement effect.
    */
   videoBackgroundImages?: VideoBackgroundImage[];
+  /* @conditional-compile-remove(video-background-effects) */
+  /**
+   * Default set of background images for background replacement effect.
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:126]: Does this also include the custom backgrounds that contoso would upload or provide to their composite?

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 2]
```diff
@@ -247,6 +252,23 @@ export interface VideoBackgroundImage {
...
+/**
+ * Contains the attibutes of a selected video background effect
+ *
+ * @beta
+ */
+export interface VideoBackgroundEffect {
+  /**
+   * Type of video background effect, can be Blur, Replacement or None.
+   */
+  type: 'Blur' | 'Replacement' | 'None';
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:265]: is replacement to represent the backgrounds that contoso would upload as well? they are able to provide their own images right? (not custom user uploads)

File: packages/react-composites/src/composites/CallComposite/utils/Utils.ts [hunk 3]
```diff
@@ -40,6 +42,27 @@ export const isCameraOn = (state: CallAdapterState): boolean => {
   return false;
 };
 
+/* @conditional-compile-remove(video-background-effects) */
+/**
+ * @private
+ */
+export const startSelectedVideoEffect = async (adapter: CommonCallAdapter): Promise<void> => {
+  if (adapter.getState().selectedVideoBackgroundEffect) {
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/CallComposite/utils/Utils.ts:50]: is there a way to write a unit test for this? should we be providing warnings to the console that there was a problem starting a effect?

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 4]
```diff
@@ -247,6 +252,23 @@ export interface VideoBackgroundImage {
...
+  /**
+   * Configuration details of selected video background effect.
+   */
+  videoEffectConfig?: BackgroundBlurConfig | BackgroundReplacementConfig;
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:269]: We shouldn't pass complex objects as part of state, state needs to be readonly interfaces only.

Instead here, can split `VideoBackgroundEffect` into three interfaces:
```ts
interface VideoBackgroundEffect = VideoBackgroundNoneEffect | VideoBackgroundBlurEffect | VideoBackgroundReplacementEffect;

interface VideoBackgroundNoneEffect  {
  effectName: 'None';
}

interface VideoBackgroundBlurEffect {
  effectName: 'BackgroundBlur';
}

interface VideoBackgroundReplacementEffect {
  effectName: 'BackgroundReplacement';
  url: string
}
```

Comment (JamesBurnside) [reply packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:269]: This change is just for the ClientState, the setSelectedVideoBackgroundEffect should probably still take the regaulr config object

File: packages/react-composites/review/beta/react-composites.api.md [hunk 5]
```diff
@@ -216,6 +216,7 @@ export type CallAdapterClientState = {
     roleHint?: Role;
     cameraStatus?: 'On' | 'Off';
     videoBackgroundImages?: VideoBackgroundImage[];
+    selectedVideoBackgroundEffect?: VideoBackgroundEffect;
```
Comment (JamesBurnside) [inline packages/react-composites/review/beta/react-composites.api.md:219]: I don't see it in this PR, but the CallComposite background picker's selected item should rely on this state now so its always consistent

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts (hunk 4 above)
Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts]: done

File: packages/react-composites/review/beta/react-composites.api.md (hunk 5 above)
Comment (prabhjot-msft) [reply packages/react-composites/review/beta/react-composites.api.md]: I am working on a bug for selected item, will make the change sin that PR https://skype.visualstudio.com/SPOOL/_workitems/edit/3217313

File: packages/communication-react/review/beta/communication-react.api.md [hunk 6]
```diff
@@ -3331,13 +3339,33 @@ export interface VerticalGalleryStyles extends BaseCustomStyles {
...
+    // (undocumented)
+    effectKey: 'None';
+}
+
+// @beta
+export interface VideoBackgroundReplacementEffect {
+    // (undocumented)
+    backgroundImageUrl: string;
+    // (undocumented)
+    effectKey: string;
```
Comment (JamesBurnside) [inline packages/communication-react/review/beta/communication-react.api.md:3366]: update `string` to be the actual name
```suggestion
    effectKey: 'Replace';
```

File: packages/communication-react/review/beta/communication-react.api.md [hunk 7]
```diff
@@ -3331,13 +3339,33 @@ export interface VerticalGalleryStyles extends BaseCustomStyles {
     controlBar?: VerticalGalleryControlBarStyles;
 }
 
+// @beta
+export interface VideoBackgroundBlurEffect {
+    // (undocumented)
+    effectKey: 'None';
```
Comment (JamesBurnside) [inline packages/communication-react/review/beta/communication-react.api.md:3345]: update key here
```suggestion
    effectKey: 'Blur';
```

File: packages/react-composites/src/composites/CallComposite/utils/Utils.ts [hunk 8]
```diff
@@ -40,6 +42,23 @@ export const isCameraOn = (state: CallAdapterState): boolean => {
...
+    const selectedVideoBackgroundEffect = adapter.getState().selectedVideoBackgroundEffect;
+    if (selectedVideoBackgroundEffect?.effectName === 'blur') {
+      await adapter.blurVideoBackground();
+    } else if (selectedVideoBackgroundEffect?.effectName === 'none') {
+      await adapter.stopVideoBackgroundEffect();
+    } else if (selectedVideoBackgroundEffect?.effectName === 'replacement') {
+      adapter.replaceVideoBackground({ backgroundImageUrl: selectedVideoBackgroundEffect.backgroundImageUrl });
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/utils/Utils.ts:57]: for consistency await this

**Review by JamesBurnside — APPROVED** [review]

File: change/@azure-communication-react-a80faf87-e35d-493e-91e9-1e9ecdeb1e43.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
```
Comment (dmceachernmsft) [inline change/@azure-communication-react-a80faf87-e35d-493e-91e9-1e9ecdeb1e43.json:2]: why do we have 2 changelogs for this PR?

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 10]
```diff
@@ -247,6 +252,64 @@ export interface VideoBackgroundImage {
...
+  | VideoBackgroundBlurEffect
+  | VideoBackgroundReplacementEffect;
+
+/* @conditional-compile-remove(video-background-effects) */
+/**
+ * Contains the attibutes to remove video background effect
+ *
+ * @beta
+ */
+export interface VideoBackgroundNoneEffect {
```
Comment (dmceachernmsft) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:272]: Do we need two types here? could we just  have `VideoBackgroundBlurEffect` where `effectName: 'blur' | 'none'` is the property?

File: change/@azure-communication-react-a80faf87-e35d-493e-91e9-1e9ecdeb1e43.json (hunk 9 above)
Comment (prabhjot-msft) [reply change/@azure-communication-react-a80faf87-e35d-493e-91e9-1e9ecdeb1e43.json:2]: I made some API changes in second patch and ran `rush changelog` by mistake, looks like extra one got generated

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts (hunk 10 above)
Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:272]: Yeah, so none effect is basically to remove any video effects that includes both `blur` and `replacement` effects. If we add `none` to `VideoBackgroundBlurEffect ` that is kind of misleading that blur that selecting none would remove blur effect, whereas we want to keep it as separate type to remove either of two effects.

**Review by dmceachernmsft — APPROVED** [review]
approving for either types.

**Pull request merged**